bridgesync/e2e_test.go

```diff
@@ -92,21 +100,38 @@ func TestBridgeEventE2E(t *testing.T) {
 
 	// Wait for syncer to catch up
 	time.Sleep(time.Second * 2) // sleeping since the processor could be up to date, but have pending reorgs
-	lb, err := setup.L1Environment.SimBackend.Client().BlockNumber(ctx)
-	require.NoError(t, err)
+
+	lb := GetFinalizedBlockNumber(t, ctx, setup.L1Environment.SimBackend.Client())
 	helpers.RequireProcessorUpdated(t, setup.L1Environment.BridgeSync, lb)
 
 	// Get bridges
 	lastBlock, err := setup.L1Environment.SimBackend.Client().BlockNumber(ctx)
 	require.NoError(t, err)
-	actualBridges, err := setup.L1Environment.BridgeSync.GetBridges(ctx, 0, lastBlock)
+	lastProcessedBlock, err := setup.L1Environment.BridgeSync.GetLastProcessedBlock(ctx)
 	require.NoError(t, err)
-
+	actualBridges, err := setup.L1Environment.BridgeSync.GetBridges(ctx, 0, lastProcessedBlock)
+	require.NoError(t, err)
+	log.Infof("lastBlockOnChain:%d lastProcessedBlock: %d, len(actualBridges): %d", lb, lastProcessedBlock, len(actualBridges))
 	// Assert bridges
 	expectedRoot, err := setup.L1Environment.BridgeContract.GetRoot(nil)
 	require.NoError(t, err)
 	root, err := setup.L1Environment.BridgeSync.GetExitRootByIndex(ctx, expectedBridges[len(expectedBridges)-1].DepositCount)
 	require.NoError(t, err)
+	log.Infof("expectedRoot: %s lastBlock: %d lastFinalized:%d DepositCount:%d ", common.Hash(expectedRoot).Hex(), lastBlock, lb, expectedBridges[len(expectedBridges)-1].DepositCount)
+	for i := 119; i >= 00; i-- {
+		root, err := setup.L1Environment.BridgeSync.GetExitRootByIndex(ctx, uint32(i))
+		require.NoError(t, err)
+		log.Infof("DepositCount:%d root: %s", i, root.Hash.Hex())
+	}
 	require.Equal(t, common.Hash(expectedRoot).Hex(), root.Hash.Hex())
 	require.Equal(t, expectedBridges, actualBridges)
 }
+
+func GetFinalizedBlockNumber(t *testing.T, ctx context.Context, client simulated.Client) uint64 {
```

Comment:
No need to export it:

```suggestion
func getFinalizedBlockNumber(t *testing.T, ctx context.Context, client simulated.Client) uint64 {
```